eclair-core/src/main/scala/fr/acinq/eclair/blockchain/fee/OnChainFeeConf.scala

```diff
@@ -62,14 +47,18 @@ case class FeerateTolerance(ratioLow: Double, ratioHigh: Double, anchorOutputMax
   }
 }
 
-case class OnChainFeeConf(feeTargets: FeeTargets, feeEstimator: FeeEstimator, spendAnchorWithoutHtlcs: Boolean, closeOnOfflineMismatch: Boolean, updateFeeMinDiffRatio: Double, private val defaultFeerateTolerance: FeerateTolerance, private val perNodeFeerateTolerance: Map[PublicKey, FeerateTolerance]) {
+case class OnChainFeeConf(feerates: AtomicReference[FeeratesPerKw], safeUtxosThreshold: Int, spendAnchorWithoutHtlcs: Boolean, closeOnOfflineMismatch: Boolean, updateFeeMinDiffRatio: Double, private val defaultFeerateTolerance: FeerateTolerance, private val perNodeFeerateTolerance: Map[PublicKey, FeerateTolerance]) {
```

Comment:
Putting `feerates` is not very pure (because the case class is now mutable), but it simplify greatly the code by containing everything related to feerates in `OnChainFeeConf`.

Unfortunately we can't make `feerates` private in order to make it testable.

Reply:
It's a bit frustrating though...everywhere we use an `OnChainFeeConf`, we actually have access to the whole `NodeParams` field, so we could keep `OnChainFeeConf` pure and pass the `feerates` separately, I'd feel more comfortable with that.

Reply:
If this contained a new kind of `FeeTargets` where each kind of transaction (funding, closing, commit, commit without htlcs) had a `slow/medium/fast` target, we could just expose `getFundingFeerate(feerates: AtomicReference[FeeratesPerKw]): FeeratePerKw` (and same for closing/commit).

Reply:
I started that way, it does make `NodeParams` nicer with a perfect symmetry between the atomic references for `blockCount` and `feerates`. However, it made the code in `Commitments` and `Helpers` quite tedious, e.g. I had to pass two args like so:
```scala
def canSendAdd(amount: MilliSatoshi, params: ChannelParams, changes: CommitmentChanges, feeConf: OnChainFeeConf, feerates: FeeratesPerKw)
```

Also, the current change doesn't make the code "less pure" because we were already storing the `FeeEstimator` inside `OnChainFeeConf`.

Reply:
> Also, the current change doesn't make the code "less pure" because we were already storing the FeeEstimator inside OnChainFeeConf.

Huh, that's true! But we now have an opportunity to make it more pure :smile: 

I think it makes more sense to have a `XXXConf` class be a really pure class, and other dynamic objects be passed separately: even for unit tests, it's simpler because the thing we'll mock is only the `AtomicRef`, right?

Reply:
> Huh, that's true! But we now have an opportunity to make it more pure 😄

Fair enough! will give it a try

> for unit tests, it's simpler because the thing we'll mock is only the AtomicRef, right?

It will be the same I think, we don't use mocking, simply directly set the atomic ref value

Reply:
Done in b8f334e7a.